src/test/java/org/broadinstitute/hellbender/engine/MultiFeatureWalkerUnitTest.java

```diff
@@ -0,0 +1,137 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.util.Log;
+import htsjdk.tribble.Feature;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MultiFeatureWalkerUnitTest extends CommandLineProgramTest {
+    private static final class DummyMultiFeatureWalker extends MultiFeatureWalker<Feature> {
+        @Argument( fullName = StandardArgumentDefinitions.FEATURE_LONG_NAME,
+                    shortName = StandardArgumentDefinitions.FEATURE_SHORT_NAME )
+        private List<FeatureInput<Feature>> featureInputs;
+
+        final List<Feature> features = new ArrayList<>();
+        @Override public void apply( Feature feature, Object header ) {
+            features.add(feature);
+        }
+    }
+
+    @Test
+    public void testDictionarySubsetIsOK() {
+        final DummyMultiFeatureWalker dummy = new DummyMultiFeatureWalker();
+        final String[] args = {
+                "--" + StandardArgumentDefinitions.VERBOSITY_NAME, Log.LogLevel.ERROR.name(),
+                // full HG38 dictionary with 3366 entries
+                "--" + StandardArgumentDefinitions.SEQUENCE_DICTIONARY_NAME, FULL_HG38_DICT,
+                // subset of HG38 dictionary with 13 entries
+                "--" + StandardArgumentDefinitions.REFERENCE_LONG_NAME, b38_reference_20_21,
+                // bam has full HG38 dictionary
+                "-" + StandardArgumentDefinitions.INPUT_SHORT_NAME, largeFileTestDir + "NA12878.alignedHg38.duplicateMarked.baseRealigned.bam",
+                // no dictionary, no sample names, with a single feature
+                "-" + StandardArgumentDefinitions.FEATURE_SHORT_NAME, packageRootTestDir + "engine/tiny_hg38.baf.txt"
+        };
```

Comment:
Consider using `ArgumentsBuilder` to build the command lines for your test cases.